explorer/models.py

```diff
@@ -147,6 +149,78 @@ class Meta:
 
 
 class QueryResult(object):
+    def get_type_code_and_column_indices_to_be_masked_dict(self):
+        """
+        Returns a dictionary of type code and column indices to be masked
+        Return type:
+        {
+            type_code: [column indices that match the type code]
+        }
+        Eg.
+        Say a table has three fields id, data, random_text. id is of type INT, data is of type JSON, and random_text is of type TEXT.
+        Then the return value will be:
+        {
+            TYPE_CODE_FOR_JSON: [1],
+            TYPE_CODE_FOR_TEXT: [2]
+        }
+        as 1 is the column index for JSON and 2 is the column index for TEXT
+        """
+        type_code_and_column_indices_to_be_masked_dict = {
+            TYPE_CODE_FOR_JSON: [],
+            TYPE_CODE_FOR_TEXT: []
+        }
+        # Collect the indices for JSON and text columns
+        for index, column in enumerate(self._description):
+            if hasattr(column, "type_code") and column.type_code in type_code_and_column_indices_to_be_masked_dict:
+                type_code_and_column_indices_to_be_masked_dict[column.type_code].append(index)
+
+        return type_code_and_column_indices_to_be_masked_dict
+
+    def mask_json_data(self, data):
+        return json.dumps(replace_regex(data, PII_MASKING_PATTERN_REPLACEMENT_DICT))
+
+    def mask_text_data(self, data):
+        return replace_regex(data, PII_MASKING_PATTERN_REPLACEMENT_DICT)
+
+    def get_masked_data(self, data, type_code):
+        """
+        Mask the data based on the type code.
+        """
+        if not data:
+            return data
+        if type_code == TYPE_CODE_FOR_JSON:
+            return self.mask_json_data(str(data))
+        elif type_code == TYPE_CODE_FOR_TEXT:
+            return self.mask_text_data(data)
+        return data
+    def mask_pii_data(self, row, type_code_and_column_indices_to_be_masked_dict):
+        """
+        Mask the JSON and TEXT data types in the row.
+        """
+        modified_row = list(row)
```

Comment:
if they are tuples, should we convert them back after our processing?

Reply:
Nope, the author was returning a list from this point onwards as well.